File: src/main/java/com/alcorstudio/ems/service/EmployeeService.java
package com.alcorstudio.ems.service;

import com.alcorstudio.ems.dto.EmployeeDto;

import java.util.List;

public interface EmployeeService {

    EmployeeDto createEmployee(EmployeeDto employeeDto);

    //GetEmployee
    EmployeeDto getEmployeeById(Long employeeId);

    //GetAllEmployees
    List<EmployeeDto> getAllEmployees();

    // Update Employee
    EmployeeDto updateEmployee(Long employeeId, EmployeeDto updatedEmployee);

    // Delete Employee

    void deleteEmployee(Long employeeId);
}

File: ems-frontend/ems-frontend-side/src/services/EmployeeService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/employees";

export const listEmployees = () => axios.get(REST_API_BASE_URL);

//Rest API for AddEmployee

export const createEmployee = (employee) =>
  axios.post(REST_API_BASE_URL, employee);

// REST Api for get method
export const getEmployee = (employeeId) =>
  axios.get(REST_API_BASE_URL + "/" + employeeId);

// REST api for put method
export const updateEmployee = (employeeId, employee) =>
  axios.put(REST_API_BASE_URL + "/" + employeeId, employee);

// Rest api for delete method

export const deleteEmployee = (employeeId) =>
  axios.delete(REST_API_BASE_URL + "/" + employeeId);
